hw/ip/otbn/dv/otbnsim/sim/sim.py

```diff
@@ -109,8 +109,12 @@ def _on_retire(self,
         pc_before = self.state.pc
         self.state.commit(sim_stalled=False)
 
-        # Fetch the next instruction
-        self._next_insn = self._fetch(self.state.pc)
+        # Stall for a cycle after any instruction with `has_fetch_stall` set.
+        if insn.has_fetch_stall:
+            self._next_insn = None
+        else:
+            # Fetch the next instruction
+            self._next_insn = self._fetch(self.state.pc)
```

Comment:
I wonder whether this would be clearer with a ternary (to avoid a reader having to understand that we're setting the same variable in both branches). Something like this?
```python3
        # Fetch the next instruction unless this instruction had
        # `has_fetch_stall` set, in which case we inject a single cycle stall.
        self._next_insn = (None if insn.has_fetch_stall
                           else self._fetch(self.state.pc))
```